server/src/main/java/org/opensearch/extensions/ExtensionsManager.java

```diff
@@ -486,15 +489,15 @@ public void beforeIndexRemoved(
                                     new IndicesModuleRequest(indexModule),
                                     acknowledgedResponseHandler
                                 );
-                                // TODO: make asynchronous
-                                inProgressIndexNameFuture.get(EXTENSION_REQUEST_WAIT_TIMEOUT, TimeUnit.SECONDS);
-                                logger.info("Received ack response from Extension");
+                                inProgressFuture.whenComplete((r, e) -> {
```

Comment:
The logic seems to be broken here: apparently the previous code was waiting for `inProgressIndexNameFuture` (the results from the action above), but it does not do that anymore.